src/ServiceControl/Infrastructure/RavenDB/Expiration/ExpiredDocumentsCleaner.cs

```diff
@@ -111,71 +92,66 @@ void Delete(PeriodicExecutor executor)
+                            }
+                            items.Add(new DeleteCommandData
+                            {
+                                Key = id
+                            });
 
-                                var bodyNotStored = doc.SelectToken("MessageMetadata.BodyNotStored", errorWhenNoMatch: false);
-                                if (bodyNotStored == null || bodyNotStored.Value<bool>() == false)
-                                {
-                                    var msgId = doc.SelectToken("MessageMetadata.MessageId", errorWhenNoMatch: false);
-                                    if (msgId != null)
-                                    {
-                                        var attachmentId = "messagebodies/" + msgId.Value<string>();
-                                        attachments.Add(attachmentId);
-                                    }
-                                }
+                            string bodyId;
+                            if (TryGetBodyId(doc, out bodyId))
+                            {
+                                attachments.Add(bodyId);
                             }
                         });
-                    }
-                    catch (OperationCanceledException)
-                    {
-                        //Ignore
-                    }
-
-                    logger.Debug("Batching deletion of {0} documents.",items.Count);
-
-                    docsToExpire += items.Count;
-                    var results = Database.Batch(items.ToArray());
-                    Database.TransactionalStorage.Batch(accessor =>
-                    {
-                        foreach (var attach in attachments)
-                        {
-                            accessor.Attachments.DeleteAttachment(attach, null);
-                        }
-                    });
-                    deletionCount = results.Count(x => x.Deleted == true);
-                    items.Clear();
                 }
-
-                if (docsToExpire == 0)
+                catch (OperationCanceledException)
                 {
-                    logger.Debug("No expired documents found");
+                    //Ignore
                 }
-                else
+
+                logger.Debug("Batching deletion of {0} documents.", items.Count);
+
+                docsToExpire += items.Count;
+                var results = database.Batch(items.ToArray());
+                database.TransactionalStorage.Batch(accessor =>
                 {
-                    logger.Debug("Deleted {0} out of {1} expired documents batch - Execution time:{2}ms", deletionCount, docsToExpire, stopwatch.ElapsedMilliseconds);
-                }
+                    foreach (var attach in attachments)
+                    {
+                        accessor.Attachments.DeleteAttachment(attach, null);
+                    }
+                });
+                deletionCount = results.Count(x => x.Deleted == true);
```

Comment:
Ah. Missed that. I thought it had been replaced by the `docsToExpire` but now I see that both are used.